testing/trino-tests/src/test/java/io/trino/tests/TestServer.java

```diff
@@ -232,6 +242,47 @@ public void testNoTransactionSupport()
 
     @Test(dataProvider = "testVersionOnErrorDataProvider")
     public void testVersionOnError(String query)
+    {
+        checkVersionOnError(query);
+    }
+
+    @DataProvider
+    public Object[][] testVersionOnErrorDataProvider()
+    {
+        return new Object[][] {
+                {"SELECT query that fails parsing"}, // fails during parsing
+                {"SELECT foo FROM no.such.table"}, // fails during analysis
+                {"SELECT 1 / 0"}, // fails during optimization
+                {"select 1 / a from (values 0) t(a)"}, // fails during execution
+        };
+    }
+
+    @Test
+    public void testVersionOnCompilerFailedError()
```

Comment:
nit add a comment why the query is expected to fail. It is not very obvious.
Or better add an assertion that it failed in expeted way.

Reply:
> Or better add an assertion that it failed in expeted way.

i forgot about that